### ISSUE TITLE
Current build permanently resets

### ISSUE THREAD

**Issue by dirkwhoffmann** [h5n1xp/Omega#29]
I've downloaded the current build, but I don't get Omega to boot any more.

The emulator resets itself periodically (which I can see in the console log): 

    Machine reset...
    PC:fc05fc6
    SP:dbfffa6
    Machine reset...
    PC:fc05fc6
    SP:dbfffa6
    ....

**Comment by h5n1xp** [comment]
Damn! I feared this would happen. I’ve been working on a new display generation algorithm, I must have left something in 😞

Let me know your Rom version and boot disk 👍🏻

**Comment by dirkwhoffmann** [comment]
I tried with: 

    ./omega ~/Dropbox/Amiga/Roms/kick12.rom  ~~/Dropbox/Amiga/Workbench/A2000\ WB\ 1.2\ D.adf 

and 

    ./omega ~/Dropbox/Amiga/Roms/kick12.rom  ~/Dropbox/Amiga/Games/Marble\ Madness.adf

**Comment by h5n1xp** [comment]
Ok, I think KS1.2 doesn’t like the slowram I have implemented, are you able to try with KS1.3 or KS2?

**Comment by dirkwhoffmann** [comment]
Yes, it's a KS 1.2 issue. With KS 1.3 ...

<img width="752" alt="Screenshot 2019-03-30 at 12 36 20" src="https://user-images.githubusercontent.com/12561945/55275581-7b895980-52e8-11e9-9f7a-e03b82112ae2.png">

Let's double click this thing 🤤

<img width="752" alt="Screenshot 2019-03-30 at 12 35 01" src="https://user-images.githubusercontent.com/12561945/55275589-8c39cf80-52e8-11e9-83e7-1b66b0a30fb8.png">

Looking good 😎

<img width="752" alt="Screenshot 2019-03-30 at 12 35 07" src="https://user-images.githubusercontent.com/12561945/55275597-b8555080-52e8-11e9-978b-20c1271f6f2d.png">

Ouuups 😲

**Comment by dirkwhoffmann** [comment]
Defender of the Crown causes a bus error 😬:

    omega ~/Dropbox/Amiga/Roms/kick13.rom  ~/Dropbox/Amiga/Games/Defender\ of\ the\ Crown_Disk1.adf 

After hitting F1: 

    0000 - DF0 Click
    0000 - DF0 Click
    0000 - DF0 Click
    Bus error: 10

Here is the crash log: 

    Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
    0   omega                         	0x00000001029642c9 sprite2chunky + 201
    1   omega                         	0x0000000102960e23 dma_execute + 1027
    2   omega                         	0x00000001029b6860 main + 1808
    3   libdyld.dylib                 	0x00007fff7de1c3d5 start + 1

Hmmm, but not always. This time, it worked (maybe an uninitialised variable somewhere): 

Let's do something useful this weekend, e.g., winning a trophy: 

<img width="752" alt="Screenshot 2019-03-30 at 13 12 12" src="https://user-images.githubusercontent.com/12561945/55275977-c22d8280-52ed-11e9-955a-239d3f5868e7.png">

<img width="752" alt="Screenshot 2019-03-30 at 13 12 33" src="https://user-images.githubusercontent.com/12561945/55275965-ba6dde00-52ed-11e9-9371-5b8d019c472e.png">

Yeah, I'm ready 😎.

<img width="752" alt="Screenshot 2019-03-30 at 13 12 37" src="https://user-images.githubusercontent.com/12561945/55275964-b5109380-52ed-11e9-905e-23a40b0ca06e.png">

Uh, wait a moment, this is damned fast ☹️. And my opponent is too much on the left.

<img width="752" alt="Screenshot 2019-03-30 at 13 12 46" src="https://user-images.githubusercontent.com/12561945/55275962-ad50ef00-52ed-11e9-8f68-ef2a73ed4dc2.png">

This is totally unfair. I would have beaten him for sure if the emulator didn't run so fast 😖.

OK, to be honest, I can't remember that I ever won the tournament on my A500 🙄.

**Comment by mithrendal** [comment]
Haha. How about raiding castles 🏰 and rescuing princesses? Is that also going much faster now? As far as I remember the princesses were drawn very beautiful.

**Comment by dirkwhoffmann** [comment]
> As far as I remember the princesses were drawn very beautiful.

Now you know why I'm working so hard on the 4x XBR upscaler in vAmiga 😎.

**Comment by mithrendal** [comment]
Yep. Truly further blurring the line between fiction and reality. I bet this is the most technical sophisticated  form  of escapism. En effect only being topped by grand literature like JRR Tolkiens Lord of the Rings.

**Comment by h5n1xp** [comment]
Great stuff! Don’t forget, F5 will switch off the On-Screen indicators and it will also capture the mouse. If you make the window full screen it makes for a very enjoyable experience.

**Comment by h5n1xp** [comment]
> <img alt="Screenshot 2019-03-30 at 12 35 07" width="752" src="https://user-images.githubusercontent.com/12561945/55275597-b8555080-52e8-11e9-978b-20c1271f6f2d.png">
> 
> Ouuups 😲

This is because my bitplane fetch assumes the bitmap is either 320 or 640 pixels wide. As soon as some game programmer makes the bitmap slightly narrower (to free up Chipram cycles) or slightly wider to fill the overscan area, then we get these “smeared” graphics.

I have an open issue about the beam position, I have tried to write better versions, but they never work as well as what I currently have. I will get it working eventually 😊
